Src/FluentAssertions/Equivalency/NestedExclusionOptionBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using FluentAssertions.Common;
+using FluentAssertions.Equivalency.Selection;
+
+namespace FluentAssertions.Equivalency
+{
+    public class NestedExclusionOptionBuilder<TExpectation, TCurrent> : EquivalencyAssertionOptions<TExpectation>
```

Comment:
I'm a bit unsure about this construction as it seems to allow applying methods from `EquivalencyAssertionOptions` on top of `NestedExclusionOptionBuilder` without taking effect.



```cs
// Arrange
var subject = new
{
    Text = "a",
    Inner = new[]
    {
        new { Number = 1 },
        new { Number = 2 },
        new { Number = 3 } // <-- should make the test fail
    }
};

var expected = new
{
    Text = "a",
    Inner = new[]
    {
        new { Number = 2 },
        new { Number = 1 }
    }
};

// Act / Assert
subject.Should().BeEquivalentTo(expected, opt => opt
    .WithStrictOrdering()
    .For(e => e.Inner)
        .WithoutStrictOrdering()
);
```

This also compiles
```cs
subject.Should().BeEquivalentTo(expected, opt => opt
    .For(e => e.Inner)
);
```
but should probably(?) just have been 

```cs
subject.Should().BeEquivalentTo(expected, opt => opt
    .Excluding(e => e.Inner)
);
```

Maybe `Restriction<T>`  which is used for the `Using<int>(ctx.Subject.Should().BeCloseTo(ctx.Expectation, 1)).WhenTypeIs<int>()` has some solutions?

Reply:
Regarding the first issue where `.For(e => e.Inner)` actually excluded the whole `Inner` collection, this is fixed. I added a test `For_Does_Not_Exclude`. The issue was that I immediately called `AddSelectionRule` in the `For` method.

Is this fine or should the following not compile at all?
```c#
subject.Should().BeEquivalentTo(expected, opt => opt
    .For(e => e.Inner)
);
```

Reply:
@dennisdoomen I made `NestedExclusionOptionBuilder` not inherit from `EquivalencyAssertionOptions`. This way a statement like this does not compile:

```c#
subject.Should().BeEquivalentTo(expected, opt => opt
    .For(e => e.Inner)
);
```